tests/components/pyload/test_config_flow.py

```diff
@@ -245,3 +247,215 @@ async def test_reconfigure_errors(
     assert result["reason"] == "reconfigure_successful"
     assert config_entry.data == USER_INPUT
     assert len(hass.config_entries.async_entries()) == 1
+
+
+async def test_hassio_discovery(
+    hass: HomeAssistant,
+    mock_setup_entry: AsyncMock,
+    mock_pyloadapi: AsyncMock,
+) -> None:
+    """Test flow started from Supervisor discovery."""
+
+    mock_pyloadapi.login.side_effect = InvalidAuth
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        data=HassioServiceInfo(
+            config=ADDON_DISCOVERY_INFO,
+            name="pyLoad-ng Addon",
+            slug="p539df76c_pyload-ng",
+            uuid="1234",
+        ),
```

Comment:
we could move `HassioServiceInfo` to a const since it is being repeated in all tests